frontend/src/metabase/query_builder/components/QuestionActions.tsx

```diff
@@ -64,25 +53,30 @@ interface Props {
   turnDatasetIntoQuestion: () => void;
   onInfoClick: () => void;
   onModelPersistenceChange: () => void;
-  isModerator: boolean;
-  softReloadCard: () => void;
 }
 
 const QuestionActions = ({
   isBookmarked,
   isShowingQuestionInfoSidebar,
-  isMetabotEnabled,
   handleBookmark,
   onOpenModal,
   question,
   setQueryBuilderMode,
   turnDatasetIntoQuestion,
   onInfoClick,
   onModelPersistenceChange,
-  isModerator,
-  softReloadCard,
 }: Props) => {
-  const bookmarkTooltip = isBookmarked ? t`Remove from bookmarks` : t`Bookmark`;
+  const isMetabotEnabled = useSelector(state =>
+    getSetting(state, "is-metabot-enabled"),
+  );
+  const isModerator = useSelector(getUserIsAdmin);
+
+  const dispatch = useDispatch();
+
+  const dispatchSoftReloadCard = useCallback(
+    () => dispatch(softReloadCard()),
+    [dispatch],
+  );
```

Comment:
If this function isn't used in other hooks' dependencies then we don't need to memoize it. I think we should only memoize functions when needed e.g. when their reference shouldn't change between renders.